src/compiler/utilities.ts

```diff
@@ -4129,13 +4151,16 @@ namespace ts {
         return stringContains(version, "-dev") || stringContains(version, "-insiders");
     }
 
-    export function createTextWriter(newLine: string): EmitTextWriter {
+    export function createTextWriter(newLine: string, indentation?: string | number): EmitTextWriter {
         let output: string;
         let indent: number;
         let lineStart: boolean;
         let lineCount: number;
         let linePos: number;
         let hasTrailingComment = false;
+        if (indentation || indentation === 0) {

```

Comment:
Hypothetically someone could try pass an empty string instead of 0 to disable indentation. Wouldn't a check for `undefined` be nicer?

Reply:
yes... it could be written as `if (indentation !== undefined) {`  and that would allow "" to be used to disable the indentation (the same as if would if setting it to a 0 (zero) ).